python/pyiceberg/io/pyarrow.py

```diff
@@ -486,6 +499,207 @@ def expression_to_pyarrow(expr: BooleanExpression) -> pc.Expression:
     return boolean_expression_visit(expr, _ConvertToArrowExpression())
 
 
+def pyarrow_to_schema(schema: pa.Schema) -> Schema:
+    visitor = _ConvertToIceberg()
+    struct_results: List[Optional[IcebergType]] = []
+    for field in schema:
+        visitor.before_field(field)
+        struct_result = visit_pyarrow(field.type, visitor)
+        visitor.after_field(field)
+        struct_results.append(struct_result)
+    return visitor.schema(schema, struct_results)
```

Comment:
Why isn't this done in the `visit_pyarrow` method? It seems incorrect to have top-level logic that must be copied into every function that uses a visitor.

Reply:
Thank you for your suggestions. I refactored it into the `visti_pyarrow`